src/test/java/net/openhft/chronicle/bytes/perf/ByteArrayReadWriteJLBH.java

```diff
@@ -0,0 +1,193 @@
+
+import static java.lang.System.setProperty;
+
+public class ByteArrayReadWriteJLBH implements JLBHTask {
+
+    public static final int SHORT_LENGTH = 37;
+    public static final int LONG_LENGTH = 1_024;
+    private static final int ARRAY_SIZE = 1024 * 50;
+    private static final byte[] BYTE_ARRAY = new byte[ARRAY_SIZE];
+    private static final int ITERATIONS = 50_000;
+
+    private final Bytes<?> bytesImpl;
+    private JLBH jlbh;
+    private NanoSampler absoluteRead;
+    private NanoSampler absoluteWrite;
+    private NanoSampler relativeRead;
+    private NanoSampler relativeWrite;
+    private NanoSampler marshalRead;
+    private NanoSampler marshalWrite;
+    private Bytes<?> targetBytes;
+    private NanoSampler copyTo;
+    private int length;
+    private long lengthOfAsciiString;
+    private String stringForLength;
+
+    public ByteArrayReadWriteJLBH(Bytes<?> bytes, int length) {
+        this.bytesImpl = bytes;
+        this.length = length;
+    }
+
+    public static void runForBytes(Bytes<?> bytes) {
+        runForBytes(bytes, LONG_LENGTH);
+    }
+
+    public static void runForBytes(Bytes<?> bytes, int length) {
+        setProperty("jvm.resource.tracing", "false");
+        new JLBH(new JLBHOptions()
+                .warmUpIterations(15_000)
+                .iterations(ITERATIONS)
+                .runs(3)
+                .recordOSJitter(false)
+                .throughput(5_000)
+                .pauseAfterWarmupMS(100)
+                .accountForCoordinatedOmission(true)
+                .jlbhTask(new ByteArrayReadWriteJLBH(bytes, length)))
+                .start();
+    }
+
+    @Override
+    public void init(JLBH jlbh) {
+        this.jlbh = jlbh;
+        bytesImpl.write8bit(stringForLength(length));
+        absoluteRead = jlbh.addProbe("absoluteRead");
+        absoluteWrite = jlbh.addProbe("absoluteWrite");
+        relativeRead = jlbh.addProbe("relativeRead");
+        relativeWrite = jlbh.addProbe("relativeWrite");
+        marshalRead = jlbh.addProbe("marshalRead");
+        marshalWrite = jlbh.addProbe("marshalWrite");
+        copyTo = jlbh.addProbe("copyTo");
+        targetBytes = Bytes.allocateDirect(length);
+        stringForLength = stringForLength(length);
+    }
+
+    @Override
+    public void complete() {
+        TeamCityHelper.teamCityStatsLastRun(getClass().getSimpleName() + "-" + bytesImpl.getClass().getSimpleName(), jlbh, ITERATIONS, System.out);
+    }
+
+    private String stringForLength(int length) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < length; i++) {
+            stringBuilder.append(charForIndex(i));
+        }
+        return stringBuilder.toString();
+    }
+
+    private static char charForIndex(int i) {
+        return (char) ('a' + i % 62);
+    }
+
+    @Override
+    public void run(long startTimeNS) {
+        populateBytes();
+        bytesImpl.read(0, BYTE_ARRAY, 0, length);
```

Comment:
There's a weird artifact where the first test is slower than the ones that follow. Putting this here was an attempt at mitigating that, by wearing that slowness on a run that wasn't timed.